server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/DefaultMirrorAccessController.java

```diff
@@ -0,0 +1,182 @@
+import com.linecorp.centraldogma.server.internal.storage.repository.CrudRepository;
+import com.linecorp.centraldogma.server.internal.storage.repository.HasRevision;
+import com.linecorp.centraldogma.server.metadata.UserAndTimestamp;
+import com.linecorp.centraldogma.server.mirror.MirrorAccessController;
+
+public class DefaultMirrorAccessController implements MirrorAccessController {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultMirrorAccessController.class);
+
+    private static final UuidGenerator idGenerator = new UuidGenerator();
+
+    @Nullable
+    private CrudRepository<MirrorAccessControl> repository;
+
+    public DefaultMirrorAccessController(CrudRepository<MirrorAccessControl> repository) {
+        this.repository = repository;
+    }
+
+    public DefaultMirrorAccessController() {}
+
+    public void setRepository(CrudRepository<MirrorAccessControl> repository) {
+        checkState(this.repository == null, "repository is already set.");
+        this.repository = repository;
+    }
+
+    private CrudRepository<MirrorAccessControl> repository() {
+        checkState(repository != null, "repository is not set.");
+        return repository;
+    }
+
+    public CompletableFuture<MirrorAccessControl> add(MirrorAccessControlRequest request, Author author) {
+        return repository().save(MirrorAccessControl.from(request, author), author)
+                           .thenApply(HasRevision::object);
+    }
+
+    public CompletableFuture<MirrorAccessControl> update(MirrorAccessControlRequest request, Author author) {
+        return repository().update(MirrorAccessControl.from(request, author), author)
+                           .thenApply(HasRevision::object);
+    }
+
+    public CompletableFuture<MirrorAccessControl> get(String id) {
+        return repository().find(id).thenApply(HasRevision::object);
+    }
+
+    public CompletableFuture<List<MirrorAccessControl>> list() {
+        return repository().findAll().thenApply(list -> list.stream()
+                                                            .map(HasRevision::object)
+                                                            .collect(toImmutableList()));
+    }
+
+    @Override
+    public CompletableFuture<Boolean> allow(String targetPattern, String reason, int order) {
+        final Author author = Author.SYSTEM;
+        final MirrorAccessControl accessControl =
+                new MirrorAccessControl(idGenerator.generateId().toString(), targetPattern, true,
+                                        reason, 0, UserAndTimestamp.of(author));
+        logger.info("Allowing the target pattern: {}", accessControl);
+        // If there is a duplicate target pattern, the order will be considered first.
+        // If the order is the same, the latest one will be considered first.
+        return repository().save(accessControl, author).thenApply(unused -> true);
```

Comment:
How can we delete this access control later? Do we just need to disallow it?

Reply:
Currently, it isn't supported by `MirrorAccessControler` API. If necessary, should we use the UI to manage it?

Considering the current usability, I thought the design is sufficient.

Reply:
I see. I was just curious because the ID is automatically generated.